torch/distributed/fsdp/_init_utils.py

```diff
@@ -488,7 +488,7 @@ def _init_param_handle_from_module(
     _check_single_device_module(fully_sharded_module, state._ignored_params, device_id)
     device_from_device_id = _get_device_from_device_id(device_id, state.rank)
     is_meta_module, is_torchdistX_deferred_init = _need_to_materialize_module(
-        fully_sharded_module, state._ignored_params
+        fully_sharded_module, state._ignored_params, state._ignored_modules
```

Comment:
this looks like a former bug too, did we add unittests here?

Reply:
We are just using ignored modules to skip their buffers. Otherwise, normally `_ignored_params` is derived from `_ignored_modules` (except for the parameter `ignored_states` path, which is a known bug).